### PR TITLE
Fixed static check failures for pkg/volume/, pkg/kubelet, pkg/util and cmd/kubelet

### LINKED ISSUE #81657: Fix static check failures
As #81189 has been merged, there are a lot of packages that cannot pass the static check and we need to clean them up!

Step 1: Remove one or two packages (related packages) from the list in - https://github.com/kubernetes/kubernetes/blame/master/hack/.staticcheck_failures
Step 2: Run hack/verify-staticcheck.sh and look at the logged errors
Step 3: Fix the code to ensure that hack/verify-staticcheck.sh runs cleanly.
Step 4: Run hack/update-bazel.sh and hack/update-gofmt.sh to make sure we fix any problems with gofmt or bazel.
Step 5: File a PR with the change

- please append the packages you picked at the end.
- For existing contributor, please make sure this list is update to date

- Don't use `Fixes: #81657` which will shut down the issue,  just use `Ref:#81657`

### PR DESCRIPTION
**Pull request by hpandeycodeit**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

Fixing Static Check Failures for the following packages

```
cmd/kubelet/app
pkg/util/rlimit/
pkg/kubelet/dockershim
pkg/volume/util/operationexecutor
pkg/volume/util/fsquota
pkg/kubelet/dockershim/libdocker
```

**Which issue(s) this PR fixes**:

Ref: #81657 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Comment by xichengliudui** [comment]
It should be ref #81657 , not Fixes #81657, which will lead to the issue closing

File: hack/.staticcheck_failures [hunk 1]
```diff
@@ -45,8 +45,6 @@ pkg/kubelet/cm/devicemanager
 pkg/kubelet/config
 pkg/kubelet/container/testing
 pkg/kubelet/dockershim/libdocker
```
Comment (RainbowMango) [inline hack/.staticcheck_failures]: Why not fix `pkg/kubelet/dockershim/libdocker` as well?

File: hack/.staticcheck_failures [hunk 2]
```diff
@@ -117,12 +115,8 @@ pkg/volume/quobyte
 pkg/volume/rbd
 pkg/volume/scaleio
 pkg/volume/storageos
-pkg/volume/util
 pkg/volume/util/fsquota
-pkg/volume/util/fsquota/common
-pkg/volume/util/nsenter
 pkg/volume/util/operationexecutor
```
Comment (RainbowMango) [inline hack/.staticcheck_failures]: Why not fix `pkg/volume/util/operationexecutor` as well.

File: hack/.staticcheck_failures [hunk 3]
```diff
@@ -117,12 +115,8 @@ pkg/volume/quobyte
 pkg/volume/rbd
 pkg/volume/scaleio
 pkg/volume/storageos
-pkg/volume/util
 pkg/volume/util/fsquota
```
Comment (RainbowMango) [inline hack/.staticcheck_failures]: Why not fix pkg/volume/util/fsquota as well?

**Review by mattjmcnaughton — COMMENTED** [review]
Thanks for your pr!

Would you mind updating the title? It doesn't feel like it fully describes all the different aspects of this pr (i.e. you fixed static check failures for more than just the kubelet).

Also, I think the kubelet specific fixes are also in https://github.com/kubernetes/kubernetes/pull/81206 (cc @tallclair), so we just want to make sure there isn't a conflict when it comes time to merge.

File: hack/.staticcheck_failures (hunk 3 above)
Comment (hpandeycodeit) [reply hack/.staticcheck_failures]: Sure. I will fix the following as well : 

```
pkg/kubelet/dockershim/libdocker
pkg/volume/util/operationexecutor
pkg/volume/util/fsquota
```

**Comment by hpandeycodeit** [comment]
The static check errors reported here are not reported in my local. Am I missing something?
@SataQiu 


```
Errors from staticcheck:
pkg/kubelet/dockershim/network/cni/cni_test.go:75:5: this value of err is never used (SA4006)
pkg/kubelet/dockershim/network/cni/cni_test.go:331:10: this value of err is never used (SA4006)
pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go:69:2: const defaultIPAMDir is unused (U1000)
pkg/volume/util/atomic_writer_test.go:756:49: argument err is overwritten before first use (SA4009)
pkg/volume/util/fsquota/common/quota_linux_common.go:37:2: const acct is unused (U1000)
pkg/volume/util/fsquota/common/quota_linux_common.go:38:2: const enforcing is unused (U1000)
pkg/volume/util/fsquota/project.go:100:5: this value of err is never used (SA4006)
pkg/volume/util/fsquota/project.go:100:21: Errorf is a pure function but its return value is ignored (SA4017)
pkg/volume/util/fsquota/project.go:103:5: this value of err is never used (SA4006)
pkg/volume/util/fsquota/project.go:103:21: Errorf is a pure function but its return value is ignored (SA4017)
pkg/volume/util/fsquota/project.go:106:4: this value of err is never used (SA4006)
pkg/volume/util/fsquota/project.go:106:20: Errorf is a pure function but its return value is ignored (SA4017)
pkg/volume/util/fsquota/project.go:168:31: identical expressions on the left and right side of the '==' operator (SA4000)
pkg/volume/util/fsquota/quota_linux.go:227:18: this value of err is never used (SA4006)
pkg/volume/util/fsquota/quota_linux.go:300:50: argument poduid is overwritten before first use (SA4009)
pkg/volume/util/fsquota/quota_linux.go:412:10: this value of err is never used (SA4006)
pkg/volume/util/fsquota/quota_linux_test.go:46:7: const dummyMountDataPquota is unused (U1000)
pkg/volume/util/fsquota/quota_linux_test.go:52:7: const dummyMountDataNoPquota is unused (U1000)
pkg/volume/util/fsquota/quota_linux_test.go:59:7: const dummyMountTest is unused (U1000)
pkg/volume/util/fsquota/quota_linux_test.go:577:16: this value of err is never used (SA4006)
pkg/volume/util/nsenter/nsenter_mount.go:112:9: empty branch (SA9003)
pkg/volume/util/subpath/subpath_linux.go:234:81: argument err is overwritten before first use (SA4009)

```

My Mac: 

```
./hack/verify-staticcheck.sh
installing staticcheck from vendor
Congratulations!  All Go source files have been checked.
```

**Comment by SataQiu** [comment]
Thanks for your contribution @hpandeycodeit 
First, you need change your PR description : `Fixes #81657` -> `Ref: #81657`
Second, I think the reason the error has not been detected is that you have not installed `staticcheck` correctly. Try to install it by:
```
$ go install honnef.co/go/tools/cmd/staticcheck
$ export PATH:${GOPATH}/bin:${PATH}
$ ./hack/verify-staticcheck.sh
```

/approve
for hack

File: pkg/kubelet/dockershim/helpers_linux.go [hunk 4]
```diff
@@ -34,6 +34,15 @@ import (
...
 
+var(
+
+	defaultSeccompOpt = []dockerOpt{{"seccomp", "unconfined", ""}}
+)
+const(
+
+	dockerNetNSFmt = "/proc/%v/ns/net"
+
+)
```
Comment (tallclair) [inline pkg/kubelet/dockershim/helpers_linux.go]: nit: too much whitespace. Prefer:
```go
var defaultSeccompOpt = []dockerOpt{{"seccomp", "unconfined", ""}}

const dockerNetNSFmt = "/proc/%v/ns/net"
```

File: pkg/volume/util/fsquota/quota.go [hunk 5]
```diff
@@ -44,6 +44,6 @@ type Interface interface {
 	ClearQuota(m mount.Interface, path string) error
 }
 
-func enabledQuotasForMonitoring() bool {
+func EnabledQuotasForMonitoring() bool {
```
Comment (tallclair) [inline pkg/volume/util/fsquota/quota.go]: Why did you export this?

Comment (hpandeycodeit) [reply pkg/volume/util/fsquota/quota.go]: staticcheck reported this function never used. However, `/pkg/volume/util/fsquota/quota_linux.go` is calling this function (same package scope)

File: cmd/kubelet/app/options/globalflags.go [hunk 6]
```diff
@@ -51,8 +51,8 @@ func normalize(s string) string {
 	return strings.Replace(s, "_", "-", -1)
 }
 
-// register adds a flag to local that targets the Value associated with the Flag named globalName in global
-func register(global *flag.FlagSet, local *pflag.FlagSet, globalName string) {
+// Register adds a flag to local that targets the Value associated with the Flag named globalName in global
+func Register(global *flag.FlagSet, local *pflag.FlagSet, globalName string) {
```
Comment (mkumatag) [inline cmd/kubelet/app/options/globalflags.go]: Why exported? If not used outside of this package scope then no need to export(CapsCase)

**Review by mkumatag — CHANGES_REQUESTED** [review]

**Comment by RainbowMango** [comment]
@hpandeycodeit 

I guess you should rebase master then fix staticcheck issues.

```
Errors from staticcheck:
pkg/volume/util/fsquota/project.go:101:5: this value of err is never used (SA4006)
pkg/volume/util/fsquota/project.go:101:21: Errorf is a pure function but its return value is ignored (SA4017)
pkg/volume/util/fsquota/project.go:103:5: this value of err is never used (SA4006)
pkg/volume/util/fsquota/project.go:103:21: Errorf is a pure function but its return value is ignored (SA4017)
pkg/volume/util/fsquota/project.go:106:4: this value of err is never used (SA4006)
pkg/volume/util/fsquota/project.go:106:20: Errorf is a pure function but its return value is ignored (SA4017)
pkg/volume/util/fsquota/quota_linux.go:303:50: argument poduid is overwritten before first use (SA4009)
```

File: pkg/volume/util/subpath/subpath_linux.go [hunk 7]
```diff
@@ -231,7 +231,7 @@ func doCleanSubPaths(mounter mount.Interface, podDir string, volumeName string)
 
 		// scan /var/lib/kubelet/pods/<uid>/volume-subpaths/<volume>/<container name>/*
 		fullContainerDirPath := filepath.Join(subPathDir, containerDir.Name())
-		err = filepath.Walk(fullContainerDirPath, func(path string, info os.FileInfo, err error) error {
+		err := filepath.Walk(fullContainerDirPath, func(path string, info os.FileInfo, initErr error) error {
```
Comment (tallclair) [inline pkg/volume/util/subpath/subpath_linux.go]:
```suggestion
		err := filepath.Walk(fullContainerDirPath, func(path string, info os.FileInfo, _ error) error {
```

File: pkg/volume/util/subpath/subpath_linux.go [hunk 8]
```diff
@@ -231,7 +231,7 @@ func doCleanSubPaths(mounter mount.Interface, podDir string, volumeName string)
...
 			if path == fullContainerDirPath {
 				// Skip top level directory
 				return nil
```
Comment (tallclair) [inline pkg/volume/util/subpath/subpath_linux.go]: (line 241) returning the error will abort filewalk and neglect sibling directories. Is that what we want?

**Comment by danwinship** [comment]
/remove-sig network
(network-related parts were obsoleted by another PR)

**Pull request closed without merging**